packages/pipeline/src/data_sources/ohlcv_external/crypto_compare.ts

```diff
@@ -92,3 +100,14 @@ export class CryptoCompareOHLCVSource {
         return R.unfold(f, pair);
     }
 }
+
+function hasData(record: CryptoCompareOHLCVRecord): boolean {
+    return (
```

Comment:
If you want to avoid listing out each property individually, you could write this as:

```ts
// A record is considered to "have data" if any value except time is non-zero
const allValuesExceptTime = R.values(R.omit(['time'], record))
const notZero = value => value !== 0;
return R.any(notZero, allValuesExceptTime);
```

Totally optional, though.

Reply:
nice!